probe-rs/src/flashing/flasher.rs

```diff
@@ -48,17 +49,20 @@ impl Operation for Verify {
 /// A structure to control the flash of an attached microchip.
 ///
 /// Once constructed it can be used to program date to the flash.
-pub(super) struct Flasher<'session> {
+pub(super) struct Flasher<'session, 'progress> {
     session: &'session mut Session,
     core_index: usize,
     flash_algorithm: FlashAlgorithm,
+    #[allow(dead_code)]
+    progress: Option<&'progress FlashProgress>,
 }
 
-impl<'session> Flasher<'session> {
+impl<'session, 'progress> Flasher<'session, 'progress> {
     pub(super) fn new(
         session: &'session mut Session,
         core_index: usize,
         raw_flash_algorithm: &RawFlashAlgorithm,
+        progress: Option<&'progress FlashProgress>,
```

Comment:
This is a bit inconsistent now. The `progress` is an argument for some functions, e.g. `program`, but it's now also stored in a field.